### PR TITLE
ref(aci): pass WorkflowJob into process_workflows

### PR DESCRIPTION
**Pull request by cathteng**
`process_workflows` is currently called with a `GroupEvent` for issue types like metric issues to fire downstream actions.

To process issue alerts, we use other elements of `PostProcessJob` in post_process to evaluate conditions. As such, we should pass `PostProcessJob` into `process_workflows`. It contains the `GroupEvent`.

`PostProcessJob` is a `TypedDict` which also has the benefit of allowing us to modify the input without having to refactor stuff that's already in place.

### CODE REVIEW

File: src/sentry/workflow_engine/handlers/condition/event_state.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+from sentry.workflow_engine.models.data_condition import Condition
+from sentry.workflow_engine.registry import condition_handler_registry
+from sentry.workflow_engine.types import DataConditionHandler
+
+from .group_event import get_nested_value
+
+
+@condition_handler_registry.register(Condition.EVENT_STATE_COMPARISON)
+class EventStateConditionHandler(DataConditionHandler[dict[str, Any]]):
```
Comment (cathteng) [inline src/sentry/workflow_engine/handlers/condition/event_state.py:11]: example using kwargs as the value

File: src/sentry/workflow_engine/processors/detector.py [hunk 2]
```diff
@@ -14,11 +14,13 @@
 
 
 # TODO - cache these by evt.group_id? :thinking:
-def get_detector_by_event(evt: GroupEvent) -> Detector:
-    issue_occurrence = evt.occurrence
+def get_detector_by_event(job: PostProcessJob) -> Detector:
```
Comment (ceorourke) [inline src/sentry/workflow_engine/processors/detector.py:17]: nit: `get_detector_by_job`?

File: src/sentry/workflow_engine/processors/detector.py [hunk 3]
```diff
@@ -14,11 +14,13 @@
...
+    issue_occurrence = job["event"].occurrence
```
Comment (ceorourke) [inline src/sentry/workflow_engine/processors/detector.py:18]: could be nice to pull this off `event = job["event"]` since you do that twice

**Review by ceorourke — APPROVED** [review]

**Comment by cathteng** [comment]
too many merge conflicts, replaced with https://github.com/getsentry/sentry/pull/82489

**Pull request closed without merging**